app/models/plan.rb

```diff
@@ -422,18 +403,54 @@ def latest_update
     (phases.pluck(:updated_at) + [updated_at]).max
   end
 
-  # the owner of the project
+  # The owner (aka :creator) of the project
   #
   # Returns User
   # Returns nil
   def owner
-    vals = Role.access_values_for(:creator)
-    User.joins(:roles)
-        .where("roles.plan_id = ? AND roles.access IN (?)", id, vals).first
+    usr_ids = Role.where(plan_id: id, active: true)
+                  .administrator
+                  .order(:created_at)
+                  .pluck(:user_id).uniq
+    User.where(id: usr_ids).first
+  end
+
+  # Creates a role for the specified user (will update the user's
+  # existing role if it already exists)
+  #
+  # Expects a User.id and access_type from the following list:
+  #  :creator, :administrator, :editor, :commenter
+  #
+  # Returns Boolean
+  def add_user!(user_id, access_type = :commenter)
+    user = User.where(id: user_id).first
+    if user.present?
+      role = Role.find_or_initialize_by(user_id: user_id, plan_id: self.id)
+
+      # Access is cumulative, so set the appropriate flags
+      # (e.g. an administrator can also edit and comment)
+      case access_type
+      when :creator
+        role.creator = true
+        role.administrator = true
+        role.editor = true
+      when :administrator
+        role.administrator = true
+        role.editor = true
+      when :editor
+        role.editor = true
+      when :reviewer
+        role.reviewer = true
+      end
+      role.commenter = true
```

Comment:
If we dont have any roles which cannot comment, do we need that designation anymore?
I suppose we might always add something like an 'observer' in which has read-only no-comment access. 
Similar to the super/org-admin view into a plan without review status, but with this access for some reason scoped to a subset of all plans.

Reply:
Yeah. Thought about the same as I was refactoring `reviewable_by?` and `commendable_by?` aside from the super/org admin roles and the Branding config stuff you could get away with `Role.where(plan_id: id, user_id: user_id, active: true).any?`.

As you point out though it gives us an opportunity to add further granularity in the future. I don't think the addition of the access level check adds much overhead right now